### PR TITLE
Hw_10

### PR DESCRIPTION
**Pull request by mewlunar**

### CODE REVIEW

File: Lesson_10_hw/task_2.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+// 1. Создайте функцию delay, принимающую на вход коллбэк функцию и количество милисекунд.
+//     Функция должна исполнить колбэк строго через переданное количество миллисекунд
+//     Пример: delay(() => console.log('hello'), 2000) // Через 2 секунды в консоли появится слово hello
+function delay(callback, time) {
+  return setTimeout(() => {
+    callback();
+  }, time);
```
Comment (anatoly-karpovich) [inline Lesson_10_hw/task_2.js]:
```suggestion
  return setTimeout(callback, time);
```

File: Lesson_10_hw/task_2.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+Promise.all([promise1, promise2]).then((values) => {
+  console.log(values);
+});
+
+Promise.allSettled([promise1, promise2]).then((values) => {
+  console.log(values);
+});
+
+async function PromisesWithAll() {
```
Comment (anatoly-karpovich) [inline Lesson_10_hw/task_2.js]:
```suggestion
async function promisesWithAll() {
```

File: Lesson_10_hw/task_2.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      promise2,
+    ]);
+    console.log(promise1Result);
+    console.log(promise2Result);
+  } catch (error) {
+    console.error(error.message);
+  }
+}
+
+async function PromisesWithAllSettled() {
```
Comment (anatoly-karpovich) [inline Lesson_10_hw/task_2.js]:
```suggestion
async function promisesWithAllSettled() {
```

File: Lesson_10_hw/task_2.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        console.log(`Rejected:`, result.reason);
+      }
+    });
+  } catch (error) {
+    console.error(error.message);
+  }
+}
+
+PromisesWithAll();
+PromisesWithAllSettled();
```
Comment (anatoly-karpovich) [inline Lesson_10_hw/task_2.js]:
```suggestion
promisesWithAll();
promisesWithAllSettled();
```

File: Lesson_10_hw/task_3.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  try {
+    const [usersResponse, albumsResponse, photosResponse] = await Promise.all([
+      fetch("https://jsonplaceholder.typicode.com/users"),
+      fetch("https://jsonplaceholder.typicode.com/albums"),
+      fetch("https://jsonplaceholder.typicode.com/photos"),
+    ]);
+
+    const users = await usersResponse.json();
+    const albums = await albumsResponse.json();
+    const photos = await photosResponse.json();
```
Comment (anatoly-karpovich) [inline Lesson_10_hw/task_3.js:39]: Can be done with promise.all too)

**Review by anatoly-karpovich — APPROVED** [review]
Awesome! Great job!

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**